code/datums/components/jammer.dm

```diff
@@ -0,0 +1,22 @@
+/datum/component/jammer
+	var/active = FALSE
+	var/range
+	var/jammer_name
+
+/datum/component/jammer/Initialize(range, var/jammer_name = parent)
+	RegisterSignal(COMSIG_ITEM_ATTACK_SELF, .proc/Toggle)
+	src.range = range
+	src.jammer_name = jammer_name
+
+/datum/component/jammer/Destroy()
+	if(active)
+		GLOB.active_jammers -= src
+
+/datum/component/jammer/proc/Toggle(mob/user)
+	to_chat(user,"<span class='notice'>You [active ? "deactivate" : "activate"] [jammer_name].</span>")
+	active = !active
+	if(active)
+		GLOB.active_jammers |= src

```

Comment:
`+=`
This component should handle itself completely, if anyone tries to add it elsewhere they're doing it wrong.

Reply:
no clue what |= does it was in the original code